ci/travis/ci.sh

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env bash
+
+{ SHELLOPTS_STACK="${SHELLOPTS_STACK-}|$(set +o); set -$-"; } 2> /dev/null  # Push caller's shell options (quietly)
```

Comment:
move this to function and call it. 
also why is this necessary?

Reply:
It's necessary because the script is sourced, and modified the caller's state (`set -e`, `set -u`, etc.), and the caller (Travis in this case) isn't always compatible with this script's flags.